src/fu-plugin.c

```diff
@@ -742,6 +771,24 @@ fu_plugin_runner_update_cleanup (FuPlugin *plugin, FuDevice *device, GError **er
 	return TRUE;
 }
 
+void
+fu_plugin_runner_device_register (FuPlugin *plugin, FuDevice *device)
+{
+	FuPluginPrivate *priv = GET_PRIVATE (plugin);
+	FuPluginDeviceRegisterFunc func = NULL;
+
+	/* not enabled */
+	if (!priv->enabled)
+		return;
+
+	/* optional */
+	g_module_symbol (priv->module, "fu_plugin_device_added", (gpointer *) &func);
```

Comment:
Yes, although I agree I'm not 100% happy with it. I can't use `fu_plugin_device_register` as that's the name of the symbol that the plugin calls itself, rather than the callback. Maybe `fu_plugin_device_registered` or `fu_plugin_register_device` although that's just the two words switched around and is kinda confusing.